ndbench-core/src/main/java/com/netflix/ndbench/core/generators/RandomStringKeyGenerator.java

```diff
@@ -31,7 +31,7 @@ public RandomStringKeyGenerator(boolean preLoadKeys, int numKeys) {
 
     @Override
     public String getNextKey() {
-        int randomKeyIndex = kRandom.nextInt(getNumKeys());
```

Comment:
Isn't using getters better than referencing them directly?

Reply:
numKeys is a final protected field of StringKeyGenerator. Also the interface of KeyGenerator does not specify a setNumKeys() method. If there is a use case where numKeys is expected to change over the lifecycle of StringKeyGenerator, then I understand using the accessor. Judging from the KeyGenerator interface, it seems like numKeys will be immutable over the lifecycle of StringKeyGenrator instances. And, the only place where the getNumKeys method was used is the StringKeyGenerator class, so perhaps lets spin this another way and remove the accessor from the interface.

Reply:
Considering the behavior of `KeyGenerator` interface, `setters` are not added. Regarding `getNumKeys()`, even though in today's implementation it is only being used in `StringKeyGenerator`, idea is to have it for other implementations of `KeyGenerator` and it also helps in testing. I am not clear on the essence of changing it to `numKeys`, are you fixing any issue for this or running into any other issues due to this?

Reply:
I had changed it in the cli PR because I noticed it was not being used otherwise. Using smaller PR like these to keep the cli PR clean. will revert here and there. thanks.